framework/src/materials/MaterialPropertyStorage.C

```diff
@@ -350,6 +350,32 @@ MaterialPropertyStorage::swapBack(MaterialData & material_data, const Elem & ele
 }
 
 bool
+MaterialPropertyStorage::needInitialized(unsigned int n_qpoints, const Elem & elem, unsigned int side)
+{
+  bool need = false;
+
+  // Has not been intialized
+  if (props()[&elem][side].size() == 0 || propsOld()[&elem][side].size() == 0) need = true;
+  if (hasOlderProperties())
+    if (propsOlder()[&elem][side].size() == 0) need = true;
+
+  // The number of q_points has changed
+  if (!need)
```

Comment:
The logic here is kind of bizarre and hard to follow.

Can you explain what you're trying to do here and maybe we can work on restructuring this.